tests/test_unit_conversion.py

```diff
@@ -1,80 +1,1235 @@
-    quantity = 1 * from_unit
-    # no source unit
-    with pytest.raises(AssertionError, match='^No unit given as source.*'):
-        simple_units_model.units.get_conversion_factor(to_unit=to_unit)
-    with pytest.raises(AssertionError, match='^No unit given as source.*'):
-        simple_units_model.units.get_conversion_factor(to_unit)
-
-    # no target unit
-    with pytest.raises(TypeError):
-        simple_units_model.units.get_conversion_factor(from_unit=from_unit)
-    # multiple sources
-    with pytest.raises(AssertionError, match='^Multiple target.*'):
-        simple_units_model.units.get_conversion_factor(to_unit, from_unit=from_unit, quantity=quantity)
-    # incorrect types
-    with pytest.raises(AssertionError, match='^from_unit must be of type pint:Unit$'):
-        simple_units_model.units.get_conversion_factor(to_unit, from_unit=quantity)
-    with pytest.raises(AssertionError, match='^quantity must be of type pint:Quantity$'):
-        simple_units_model.units.get_conversion_factor(to_unit, quantity=from_unit)
-    with pytest.raises(AssertionError, match='^expression must be of type Sympy expression$'):
-        simple_units_model.units.get_conversion_factor(to_unit, expression=quantity)
-
-    # unit to unit
-    assert simple_units_model.units.get_conversion_factor(to_unit=to_unit, from_unit=from_unit) == 1000
-    # quantity to unit
-    assert simple_units_model.units.get_conversion_factor(to_unit=to_unit, quantity=quantity) == 1000
-    # expression to unit
-    assert simple_units_model.units.get_conversion_factor(to_unit=to_unit, expression=expression) == 1000
-
-
-def test_conversion_factor_same_units(simple_units_model):
-    """ Tests Units.get_conversion_factor() function when units are same
-    and conversion factor should be '1'. """
-    symbol_b = simple_units_model.get_symbol_by_cmeta_id("b")
-    equation = simple_units_model.get_equations_for([symbol_b])
-    expression = equation[1].rhs
-    to_unit = simple_units_model.units.ureg('per_ms').units
-    from_unit = simple_units_model.units.summarise_units(expression)
-    quantity = 1 * from_unit
-    # quantity to unit
-    assert simple_units_model.units.get_conversion_factor(to_unit=to_unit, quantity=quantity) == 1
-    # unit to unit
-    assert simple_units_model.units.get_conversion_factor(to_unit=to_unit, from_unit=from_unit) == 1
-    # expression to unit
-    assert simple_units_model.units.get_conversion_factor(to_unit=to_unit, expression=expression) == 1
+class TestUnitConversion:
+    ###############################################################
+    # fixtures
+
+    @pytest.fixture
+    def local_model(scope='function'):
+        """ Fixture to load a local copy of  the basic_ode model that may get modified. """
+        return shared.load_model('basic_ode')
+
+    @pytest.fixture
+    def model_missing_units(scope='function'):
+        """ Fixture to load a local copy of  the basic_ode model that may get modified. """
```

Comment:
These fixtures load different models...

Reply:
Yes, because I need different models AND I am going to change them so they need local scope. Using the function scope does seem to mean I get a fresh version each function. I could try and make one model that got me all the examples I need but because I check for no change as well as if there is - this impacts all my tests so I went with multiple fixtures. Is this an issue - it hasnt given me any problems ?

Reply:
Not an issue at all - I was merely commenting on the mismatch between the docstring and fixture name. Sorry for being unclear!